### DISCUSSION TITLE
beamnrc 'COMPILATION FAILURE' pop-up

### DISCUSSION THREAD

**Discussion by Kelvin-Ngowi** [nrc-cnrc/EGSnrc#1020] [Questions] [answered]
I have several times installed an egsnrc in my HP VICTUS laptop installed with a Microsoft windows 11 OS. The installation and building processes seem to have successfully completed but, things don't work in an expected way as it is supposed to just when I have to compile either using terminal or GUI. The following message pops-up.

**_D:\EGS_HOME\BEAM_EX16MVp> mingw32-make.exe
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
process_begin: CreateProcess(NULL, echo ., ...) **failed.**
make (e=2): The system cannot find the file specified.
mingw32-make: *** [D:\EGSnrc\HEN_HOUSE\makefiles\beam_makefile:94: D:\EGS_HOME\bin\win6432\BEAM_EX16MVp.exe] Error 2**_

I firstly thought may be the problem was the make tool version that i used wasn't the one required so i had to install the correct one and update the environment, but still no changes. 
According to the messages am getting it seems like the make tool can't reach or find the target file which has to be compiled.
YET i can't find the right solution for the problem.
Any help from you i would really appreciate.

[config_win6432.log](https://github.com/nrc-cnrc/EGSnrc/files/11814416/config_win6432.log)

Regards, 
Kelvin Ngowi
MSc Physics Student 
University of Dar es salaam, Tanzania

**Comment by ftessier** [comment]
My first recommendation is to install [Git for Windows](https://gitforwindows.org/), then `git clone` the EGSnrc repository and install.

Otherwise, you can check if #972 solves the issue. We are about to release a `2023a` version addressing a few such problems.

**Reply by Kelvin-Ngowi** [reply]
i did install the latest version of git for windows each time i installed the program but nothing changed. 
is it possible that the old versions of the egsnrc like 2018 and below would be running without errors?

**Reply by Kelvin-Ngowi** [reply]
can i try the installation for the older versions?

**Reply by rtownson** [reply]
Maybe this is a Windows 11 issue, I don't believe we have tested on it.

Before trying older versions, I would try the following. Delete or rename your old installation first, and navigate to `D:` on the command line:
```
git clone git@github.com:nrc-cnrc/EGSnrc.git
cd EGSnrc
git checkout fix-win-compile
```
Then run the GUI installer, checking the checkbox for setting the environment variables. Then open `HEN_HOUSE/specs/egspp_win3264.conf`. At the end of the file, see the line fortran_libs, and make sure it reads the following. If not, edit it, save the file, and try compiling your application (BEAM_EX16MVp):
```
fortran_libs = -lgfortran -lquadmath
```

Find older versions here (or clone and then use `git checkout v2021`): https://github.com/nrc-cnrc/EGSnrc/releases

Finally, consider installing linux in a virtual machine instead: https://github.com/nrc-cnrc/EGSnrc/wiki/Install-EGSnrc-on-VirtualBox

**Reply by Kelvin-Ngowi** [reply]
I have tried the above possible solutions but unfortunately I haven't successful make it. 
I was asking if there's a way I can completely remove egsnrc from my device and start  a fresh installation like nothing has ever been done.

**Reply by rtownson** [reply]
Yes, just delete the EGSnrc folder to uninstall. You could also remove the related environment variables (anything that points inside the EGSnrc directory), and the desktop shortcuts to be thorough.

**Reply by Kelvin-Ngowi** [reply]
Which installation version do you advice that I should install

**Reply by rtownson** [reply]
I'm not sure exactly what you've tried or what went wrong - so I would start with the instructions I provided above (cloning and checking out fix-win-compile, running the installer, etc). This installs the 2023 release, with some windows fixes included. Then if you can't compile an application, post your installation log, found in HEN_HOUSE/log.

**Reply by Kelvin-Ngowi** [reply]
Thanks for the response sir.
I hope this time around things might work perfectly.

Regards

**Reply by Kelvin-Ngowi** [reply]
[config_win3264.log](https://github.com/nrc-cnrc/EGSnrc/files/11977127/config_win3264.log)
hi mr townson  and tessier,
still i can't compile using egsnrc and I have attached my installation log file. please help me.

**Reply by rtownson** [reply, marked as answer]
It looks like there are still warnings about it not being a .git repository, which means you did not use `git clone` as suggested. Can you confirm that you followed my above instructions, and then double check that the installer is running in the right directory, the one that you created with the `git clone` command? If that is the case, then what happens if you navigate to the directory on the command line, and type `git status`?

**Reply by Kelvin-Ngowi** [reply]
i would really like to thank you for a help.
the compilation is now successful.
